x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/theme_darkmode_hook.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import type { SecurityPluginStart } from '@kbn/security-plugin/public';
+import { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+
+interface Deps {
+  uiSettingsClient: IUiSettingsClient;
+  security: SecurityPluginStart;
+}
+
+export const useThemeDarkmodeToggle = ({ uiSettingsClient, security }: Deps) => {
```

Comment:
Should the toggle be disabled/not shown if the `uiSettings.overrides.theme:darkMode` is set? Similar to the control son UserProfiles and Adv. Settings

Reply:
I am not sure why we would do that? Why don't we want to allow a user to set their own preference for dark mode and "force" it from at the admin level?

Reply:
@sebelga `uiSettings.overrides.theme:darkMode` is an existing config setting. I think Kurt's just suggesting hiding or disabling the toggle only if this config setting is being used, just for consistency. If the override setting is active, the per-user setting is not applicable. Disabling or hiding the toggle lets the user know.

Reply:
I don't know the reason, that is how it worked on Advanced Settings, so I followed that precedence on User Profiles.

Currently the darkMode value set in the user settings does not precede the config value, so the toggle looks misleading. Either the precedence needs to be changed or the toggle should indicate it is being overridden 

<img width="381" alt="Screenshot 2023-07-10 at 10 28 01 AM" src="https://github.com/elastic/kibana/assets/21210601/1a3f3c1f-6cd4-4c7f-9b3d-89a6d332d0ff">

Reply:
So... we build a system to allow user to customise their preference on how they would like to see Kibana (maybe they have accesibility preference). Dark mode (probably when they work at night) and light mode (in day light). And we are letting administrator decide what is best for a specific user at a specific time of day?

That does not make sense to me. 

But OK, I will align with that odd behaviour.

Reply:
I've updated the logic and don't display the dark mode toggle in the menu if it has been overridden in `kibana.yml`

https://github.com/elastic/kibana/pull/161214/commits/1015305abcf224fea6f2a9b4dad95f59cb9905b5